base/src/test/java/io/spine/util/Preconditions2Test.java

```diff
@@ -20,44 +20,145 @@
+
             assertThrowsOn("  ");
+            assertThrowsWithMessage("  ", randomString());
+
             assertThrowsOn("   ");
+            assertThrowsWithMessage("   ", randomString());
+        }
+    }
+
+    @Nested
+    @DisplayName("Check that a value is positive")
+    class PositiveValue {
+
+        void assertThrowsOn(long value) {
+            IllegalArgumentException exception =
+                    assertThrows(IllegalArgumentException.class,
+                                 () -> checkPositive(value));
+            assertThat(exception)
+                    .hasMessageThat()
+                    .contains(String.valueOf(value));
+        }
+
+        void assertThrowsWithMessage(long value, String errorMessage) {
+            IllegalArgumentException exception =
+                    assertThrows(IllegalArgumentException.class,
+                                 () -> checkPositive(value, errorMessage));
+            assertThat(exception).hasMessageThat()
+                                 .contains(errorMessage);
+        }
+
+        @Test
+        @DisplayName("rejecting zero")
+        void zero() {
+            assertThrowsOn(0);
+            assertThrowsWithMessage(0, randomString());
+        }
+
+        @Test
+        @DisplayName("rejecting negative values")
+        void negative() {
+            assertThrowsOn(-1);
+            assertThrowsWithMessage(-100, randomString());
+        }
+
+        @Test
+        @DisplayName("accepting and returning it")
+        void positive() {
+            long expected = TestValues.longRandom(1, 100_000);
+            assertThat(checkPositive(expected))
+                    .isEqualTo(expected);
+        }
+    }
+
+    @Test
+    @DisplayName("throw if checked value out of bounds")
+    void throwExceptionIfCheckedValueOutOfBounds() {
+        assertThrows(IllegalArgumentException.class,
+                     () -> checkBounds(10, "checked value", -5, 9));
+    }
+
+    @Nested
+    @DisplayName("Check that a message is not in the default state")
+    class NotDefaultState {
+
+        @Test
+        @DisplayName("throwing `IllegalStateException` for state transition checks")
+        void stateChecking() {
+            assertThrows(IllegalStateException.class,
+                         () -> checkNotDefaultState(StringValue.getDefaultInstance()));
+        }
+
+        @Test
+        @DisplayName("throwing `IllegalArgumentException` for argument checks")
+        void argumentChecking() {
+            assertThrows(IllegalStateException.class,
+                         () -> checkNotDefaultState(StringValue.getDefaultInstance()));
+        }
+
+        @Test
+        @DisplayName("return non-default value on check")
+        void returnValue() {
+            StringValue nonDefault = newUuidValue();
+            assertEquals(nonDefault, checkNotDefaultArg(nonDefault));
+            assertEquals(nonDefault, checkNotDefaultState(nonDefault, "with error message"));
```

Comment:
shouldn't it be a separate test case? we're testing a different function

Reply:
This suite test both `state` and `arg` functions. I've added more checks, it should look clearer now.